client/src/components/organisms/RequestGroupForm.tsx

```diff
@@ -0,0 +1,351 @@
+    }
+    return "";
+  }
+
+  const onNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    event.preventDefault();
+    const newName: string = event.target.value;
+
+    setChangeMade(true);
+    setName(newName);
+    setNameError(getNameError(newName))
+  }
+
+  const onAddRequestType = (newRequestTypeName: string) => {
+    const error = getInputRequestTypeNameError(newRequestTypeName);
+
+    if (error === "") {
+      setRequestTypeNames(requestTypeNames.concat([newRequestTypeName]));
+    }
+    setChangeMade(true);
+    setRequestTypesError(error);
+  }
+
+  const onDeleteRequestType = (targetRequestTypeName: string) => {
+    const newRequestTypeNames = requestTypeNames.filter((requestTypeName) => requestTypeName !== targetRequestTypeName);
+
+    setChangeMade(true);
+    setRequestTypeNames(newRequestTypeNames);
+    setRequestTypesError(getRequestTypeNamesError(newRequestTypeNames))
+  }
+
+  const onInputRequestTypeNameChange = (requestTypeName: string) => {
+    setChangeMade(true);
+    setRequestTypesError(getInputRequestTypeNameError(requestTypeName))
+    return true;
+  }
+
+  const onDescriptionChange = (newDescription: string) => {
+    setDescription(newDescription)
+    setDescriptionError(getDescriptionError(description))
+  }
+
+  const onImageChange = (newImage: string) => {
+    setImageError(getImageError(newImage))
+    if (images.find((imageUrl) => imageUrl === newImage)) {
+      setImage(newImage)
+    }
+  }
+
+  const onSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+    const tempNameError = getNameError(name);
+    const tempDescriptionError = getDescriptionError(description);
+    const tempImageError = getImageError(image);
+    const tempRequestTypesError = getRequestTypeNamesError(requestTypeNames);
+
+    if (tempNameError === "" && tempDescriptionError === "" && tempImageError === "" && tempRequestTypesError === "") {
+      // do mutation
+
+      props.handleClose()
+    }
+    else {
+      setNameError(tempNameError);
+      setDescriptionError(tempDescriptionError);
+      setImageError(tempImageError);
+
+      setRequestTypesError(tempRequestTypesError);
+    }
+  }
+
+  const handleClose = () => {
+    // id: ID
+    // name: String
+    // description: String
+    // image: String
+    // requestTypes: [ID]
+    if (changeMade) {
+      setShowAlertDialog(true);
+    }
+    else {
+      props.handleClose();
+    }
+  }
+
+
+  return <div className="request-group-form">
+    <FormModal
+      class="request-group-form-modal"
+      show={true}
+      handleClose={handleClose}
+      title={props.operation === "create" ? "Create Request Group" : "Edit Request Group"}
+      size="large">
+      {showAlertDialog &&
+        <AlertDialog
+          dialogText="You have unsaved changes to this group."
+          onExit={props.handleClose}
+          onStay={() => { setShowAlertDialog(false) }} />
+      }
+      <form onSubmit={onSubmit}>
```

Comment:
form can have className of the div inside it, just to remove the div just inside it

Reply:
are we trying to move the class `request-group-form-modal-content` to the form element?
because the `request-group-form-modal-content` div has a sibling div (`request-group-form-modal-footer`) inside the form

Reply:
true